standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -10234,93 +10234,105 @@ protected Integer getJdoResult(GetHelper<Integer> ctx) throws MetaException, NoS
   public boolean deletePartitionColumnStatistics(String catName, String dbName, String tableName,
       String partName, List<String> partVals, String colName, String engine)
       throws NoSuchObjectException, MetaException, InvalidObjectException, InvalidInputException {
+    List<String> colNames = new ArrayList<>();
+    if (colName != null) {
+      colNames.add(colName);
+    }
+    return deletePartitionMultiColumnStatistics(catName, dbName, tableName, partName, partVals, colNames, engine);
```

Comment:
The direct sql path is missing, we'd better pushing the `colNames` into the direct sql

Reply:
Let's discuss this today. I am not sure if I understand this.

Reply:
we can add the direct sql implementation for dropping